functions/src/fetch_friend_versions.ts

```diff
@@ -0,0 +1,85 @@
+import admin from "./firebase";
+import * as functions from "firebase-functions";
+import * as cors from "cors";
+import { apiError } from "./api";
+
+const friendsCollection = admin.firestore().collection("friends") as FirebaseFirestore.CollectionReference<FriendsData>;
+const auth = admin.auth();
+const schedulesCollection = admin.firestore().collection("schedules-dev");
+
+const corsHandler = cors({ origin: true });
+
+interface FriendsData {
+  terms: Record<string, Record<string, string[]>>[];
+}
+
+type VersionReturn = Record<string, string[]>;
+
+
+export const fetchFriendVersions = functions.https.onRequest(
+  async (request, response) => {
+    corsHandler(request, response, async () => {
+      const friends = request.body.friends;
+      if (friends == null || friends.length === 0) {
+        response.status(200).json({});
+        return;
+      }
+
+      const IDToken = request.body.IDToken;
+      if (IDToken == null) {
+        response.status(400).json(apiError("IDToken not provided"));
+        return;
+      }
+
+      // Authenticate token id
+      auth
+        .verifyIdToken(IDToken)
+        .then(async (decodedToken: admin.auth.DecodedIdToken) => {
+          const uid = decodedToken.uid;
+          // Fetch user's friends
+          let userFriendData : FriendsData = await friendsCollection
+            .doc(uid)
+            .get()
+            .then((doc) => doc.data()) as FriendsData;
```

Comment:
If we're already providing the friend's uid and the needed versions, why do we need to access `friendsCollection` again? Can't we just use the provided uid and versions to fetch schedules?

Reply:
that is to verify that the request body does not include any friends' schedules that the user does not have access to.